delphi-frontend/src/test/java/au/com/integradev/delphi/type/generic/constraint/RecordConstraintTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2025 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.type.generic.constraint;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.sonar.plugins.communitydelphi.api.type.StructKind.CLASS;
+import static org.sonar.plugins.communitydelphi.api.type.StructKind.RECORD;
+
+import au.com.integradev.delphi.type.factory.TypeFactoryImpl;
+import au.com.integradev.delphi.type.generic.TypeParameterTypeImpl;
+import au.com.integradev.delphi.utils.types.TypeFactoryUtils;
+import au.com.integradev.delphi.utils.types.TypeMocker;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.sonar.plugins.communitydelphi.api.symbol.scope.DelphiScope;
+import org.sonar.plugins.communitydelphi.api.type.Constraint;
+import org.sonar.plugins.communitydelphi.api.type.IntrinsicType;
+import org.sonar.plugins.communitydelphi.api.type.Type;
+import org.sonar.plugins.communitydelphi.api.type.TypeFactory;
+
+class RecordConstraintTest {
+  private static final TypeFactory FACTORY = TypeFactoryUtils.defaultFactory();
+
+  private static class SatisfiedArgumentsProvider implements ArgumentsProvider {
+    @Override
+    public Stream<Arguments> provideArguments(ExtensionContext context) {
+      Type integer = FACTORY.getIntrinsic(IntrinsicType.INTEGER);
+      Type enumeration = ((TypeFactoryImpl) FACTORY).enumeration("", DelphiScope.unknownScope());
+      return Stream.of(
+          Arguments.of(integer),
+          Arguments.of(enumeration),
+          Arguments.of(FACTORY.subrange("", integer)),
+          Arguments.of(FACTORY.subrange("", enumeration)),
+          Arguments.of(FACTORY.getIntrinsic(IntrinsicType.ANSICHAR)),
+          Arguments.of(FACTORY.getIntrinsic(IntrinsicType.WIDECHAR)),
```

Comment:
Add entries for booleans and real types.

Reply:
Done.